.bazelrc

```diff
@@ -29,3 +29,12 @@ build --http_timeout_scaling=5.0
 # This workaround is due to an incompatibility of
 # bazel_common/tools/maven/pom_file.bzl with Bazel 1.0
 build --incompatible_depset_is_not_iterable=false
+
+# Thread sanitizer configuration:
+build:tsan --strip=never
+build:tsan --copt -fsanitize=thread
+build:tsan --copt -DTHREAD_SANITIZER
+build:tsan --copt -O2
+build:tsan --copt -g
+build:tsan --copt -fno-omit-frame-pointer
+build:tsan --linkopt -fsanitize=thread
```

Comment:
(Moving my [comment](https://github.com/ray-project/ray/pull/6104#pullrequestreview-337468747) here.)

Instead of modifying `.bazelrc`, can we modify `COPTS` flags in the relevant build files? You can use conditionals there, and this seems like a perfect use case for it.  
`.bazelrc` really seems like a last resort for modifying build flags; we really shouldn't proliferate flags in it if we can avoid it. It becomes kind of a mess to keep maintained and it isn't modular across projects (not to mention it will make it harder for others to depend on us too).

Reply:
This is the standard way to include tsan with bazel, see also https://github.com/grpc/grpc/blob/master/tools/bazel.rc#L62 so we should stick with it :)

Reply:
Oh I see. I guess it makes sense if you explicitly want these `copt`s to be cross-cutting across the entire build and specifically _not_ affect those who depend on the library. Hadn't thought about it that way!